internal/graphapi/annotationnamespace.resolvers.go

```diff
@@ -56,50 +98,97 @@ func (r *mutationResolver) AnnotationNamespaceDelete(ctx context.Context, id gid
 				// TODO - :bug: - must delete one-by-one to ensure the deleted ID is available when the delete eventhook is triggered
 				// r.client.Annotation.Delete().Where(annotation.AnnotationNamespaceID(id)).Exec(ctx)
 				if err := tx.Annotation.DeleteOneID(ant.ID).Exec(ctx); err != nil {
-					return nil, err
+					logger.Errorw("failed to delete annotation", "error", err)
+					return nil, ErrInternalServerError
 				}
 				antCount++
 			}
 		} else {
-			return nil, fmt.Errorf("annotation namespace is in use and can't be deleted")
+			return nil, ErrNamespaceInUse
 		}
 	}
 
 	if err := tx.AnnotationNamespace.DeleteOneID(id).Exec(ctx); err != nil {
-		return nil, err
+		logger.Errorw("failed to delete annotation namespace", "error", err)
+		return nil, ErrInternalServerError
 	}
 
 	if err := tx.Commit(); err != nil {
-		return nil, err
+		logger.Errorw("failed to commit transaction", "error", err)
+		return nil, ErrInternalServerError
 	}
 
 	return &AnnotationNamespaceDeletePayload{DeletedID: id, AnnotationDeletedCount: antCount}, nil
 }
 
 // AnnotationNamespaceUpdate is the resolver for the annotationNamespaceUpdate field.
 func (r *mutationResolver) AnnotationNamespaceUpdate(ctx context.Context, id gidx.PrefixedID, input generated.UpdateAnnotationNamespaceInput) (*AnnotationNamespaceUpdatePayload, error) {
-	if err := permissions.CheckAccess(ctx, id, actionMetadataAnnotationNamespaceUpdate); err != nil {
-		return nil, err
+	logger := r.logger.With("annotationNamespaceID", id)
+
+	if id == "" {
+		return nil, NewInvalidFieldError("id", ErrFieldEmpty)
+	}
+
+	if *input.Name == "" {
```

Comment:
should we check if `Name` is `nil` as well? (and similarly with `statusnamespace.resolvers.go`)